lib/sparse/src/common/sparse_vector.rs

```diff
@@ -32,14 +44,81 @@ impl SparseVector {
         score
     }
 }
-impl From<Vec<(i32, f64)>> for SparseVector {
-    fn from(v: Vec<(i32, f64)>) -> Self {
-        let mut indices = Vec::with_capacity(v.len());
-        let mut values = Vec::with_capacity(v.len());
-        for (i, w) in v {
+impl TryFrom<Vec<(i32, f64)>> for SparseVector {
+    type Error = ValidationErrors;
+
+    fn try_from(tuples: Vec<(i32, f64)>) -> Result<Self, Self::Error> {
+        let mut indices = Vec::with_capacity(tuples.len());
+        let mut values = Vec::with_capacity(tuples.len());
+        for (i, w) in tuples {
             indices.push(i as u32);
             values.push(w as f32);
         }
-        SparseVector { indices, values }
+        SparseVector::new(indices, values)
+    }
+}
+
+impl Validate for SparseVector {
+    fn validate(&self) -> Result<(), ValidationErrors> {
+        let mut errors = ValidationErrors::default();
+
+        if self.indices.is_empty() {
+            errors.add("indices", ValidationError::new("must be non-empty"));
+        }
+        if self.values.is_empty() {
+            errors.add("values", ValidationError::new("must be non-empty"));
+        }
+        if self.indices.len() != self.values.len() {
+            errors.add(
+                "values",
+                ValidationError::new("must be the same length as indices"),
+            );
+        }
+        if self.indices.len() != self.indices.iter().unique().count() {
```

Comment:
I propose to remove this check because `unique` is not cheap and next check cover this one

Reply:
I decided to introduce instead a cheaper linear check and keep it separated from the sorting check.
This makes a better UX.
Also added a bunch of tests :innocent: 
https://github.com/qdrant/qdrant/pull/2965/commits/ca9f868fe9c216e814aa00f4fca5801f864b68ab